### ISSUE TITLE
How to test on ICDAR2015 training set

### ISSUE THREAD

**Issue by wjp0408** [lvpengyuan/masktextspotter.caffe2#21]
Hi, 
If  I want to output detection results on ICDAR2015 training set, what should I do ?
I change "icdar2015_test" to "icdar2015_train" in config yaml file, but i got this error:
![image](https://user-images.githubusercontent.com/34024933/54087306-907b6880-438c-11e9-96b9-5a124cf45035.png)

So why this happened...  And what should I do ?
Thanks ：）

**Comment by wjp0408** [comment]
I have solved this problem, and It caused by _charboxes = []._ 
Sorry to bother that.

**Comment by DecentMakeover** [comment]
@wjp0408 
Hi can you tell me how you solved this problem, even i'm facing a similar issue.

Thanks in advance.

**Comment by wjp0408** [comment]
> @wjp0408
> Hi can you tell me how you solved this problem, even i'm facing a similar issue.
> 
> Thanks in advance.

Just  modify the code in the end of function` load_gt_from_txt(self, gt_path, height, width)` in `lib/datasets/text_dataset.py`  like this:
  
```
if len(boxes) > 0:
            if self.use_charann:
                #print(charboxes)
                if not len(charboxes): charboxes = np.zeros((0, 10), dtype=np.float32)
                else: charboxes = np.vstack(charboxes)
                return words, np.array(boxes), np.array(polygons), charboxes, np.array(seg_areas), segmentations
            else:
                charbbs = np.zeros((0, 10), dtype=np.float32)
                return words, np.array(boxes), np.array(polygons), charbbs, np.array(seg_areas), segmentations
else:
            return [], np.zeros((0, 4), dtype=np.float32), np.zeros((0, 8), dtype=np.float32), np.zeros((0, 10), dtype=np.float32), np.zeros((0), dtype=np.float32), []
```

**Comment by DecentMakeover** [comment]
Hi
Thank you so much for your time, this works. But since i want to train the network,this will result in Loss in Nan and then exit the training script.

I'm guessing this is because the charboxes are empty, is there a way around this?

Thanks again!

**Comment by zhengjiawen** [comment]
Hi
@DecentMakeover 
I also encountered the same problem that loss is Nan, do you have any ways?
Thanks!

**Comment by zhoujianwen** [comment]
@zhengjiawen I still have no way to train the model,and I have been reporting errors.Have you solved it yet?

**Comment by DecentMakeover** [comment]
@zhengjiawen Ah, My best guess is that your data does not have character level annotation.Can you check?

**Comment by zhoujianwen** [comment]
@DecentMakeover Have you ever encountered such mistakes when training models? 
#34

**Comment by zhengjiawen** [comment]
> @zhengjiawen Ah, My best guess is that your data does not have character level annotation.Can you check?

Thanks a lot, but I train on ICDAR2015 train dataset. The problem does not reappear when I download pre-trained model.

**Comment by DecentMakeover** [comment]
but icdar 2015 does not have character level annotation right? only icdar2013 does. 
How are you training it? How are the results?

**Comment by zhengjiawen** [comment]
> @zhengjiawen I still have no way to train the model,and I have been reporting errors.Have you solved it yet?

你可以试试自己下载预训练模型，而不是使用配置文件里的url

**Comment by zhengjiawen** [comment]
> but icdar 2015 does not have character level annotation right? only icdar2013 does.
> How are you training it? How are the results?

Yes，I only get coordinate of bounding box, and use crnn to recognition.The result is not good.:)

**Comment by DecentMakeover** [comment]
Oh ,okay. What do you think of textboxesplusplus?

**Comment by zhengjiawen** [comment]
> Oh ,okay. What do you think of textboxesplusplus?

Sorry, I haven;t read the paper yet. But I think FOTS is efficient enough and performance better.

**Comment by DecentMakeover** [comment]
do you know of a good implementation? i am currently looking at this one -https://github.com/Pay20Y/FOTS_TF

**Comment by zhoujianwen** [comment]
@zhengjiawen Thank you. Which pre-training model do you use?

**Comment by zhengjiawen** [comment]
> do you know of a good implementation? i am currently looking at this one -https://github.com/Pay20Y/FOTS_TF

I am also looking for a good open source implementation. :)

**Comment by zhengjiawen** [comment]
> @zhengjiawen Thank you. Which pre-training model do you use?

You can download the pre-training model directly.
https://dl.fbaipublicfiles.com/detectron/ImageNetPretrained/MSRA/R-50.pkl

**Comment by zhoujianwen** [comment]
Thank you very much. I'll try.

**Comment by zhoujianwen** [comment]
@zhengjiawen Thank you very much. I'll try.

**Comment by zhoujianwen** [comment]
@wjp0408 
Hello, wjp0408! How to solve charboxes=[] and no error in training model?

**Comment by zhoujianwen** [comment]
> > @zhengjiawen Thank you. Which pre-training model do you use?
> 
> You can download the pre-training model directly.
> https://dl.fbaipublicfiles.com/detectron/ImageNetPretrained/MSRA/R-50.pkl

![image](https://user-images.githubusercontent.com/13137730/56688373-26840c00-670b-11e9-9f5e-82053077d955.png)

Also this error message, how do you solve it?

**Comment by zhoujianwen** [comment]
> > > @zhengjiawen Thank you. Which pre-training model do you use?
> > 
> > 
> > You can download the pre-training model directly.
> > https://dl.fbaipublicfiles.com/detectron/ImageNetPretrained/MSRA/R-50.pkl
> 
> ![image](https://user-images.githubusercontent.com/13137730/56688373-26840c00-670b-11e9-9f5e-82053077d955.png)
> 
> Also this error message, how do you solve it?

it's solved.thanks.

**Comment by ft-sunshine** [comment]
> do you know of a good implementation? i am currently looking at this one -https://github.com/Pay20Y/FOTS_TF

hi, 
can this implemen work?
thanks in advance

**Comment by DecentMakeover** [comment]
The results are not good
